### PR TITLE
Added RequireSudoOnUnix tags to PowerShellGet tests and unmark as pending

### PR DESCRIPTION
**Pull request by RDIL**
# PR Summary
Enables the test with the new tag, unmark as pending

## PR Context
Enables the check by adding `RequireSudoOnUnix` tag to `Describe` block

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [X] [Add `[feature]` to your commit messages if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by markekraus — APPROVED** [review]
LGTM

**Review by iSazonov — APPROVED** [review]

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 [hunk 1]
```diff
@@ -228,7 +228,7 @@ Describe "PowerShellGet - Script tests (Admin)" -tags @('Feature', 'RequireAdmin
         Remove-InstalledScripts
     }
 
-    ## Marked as 'Pending' on Linux for now because the test requires root privilege but we cannot do it now in our Travis CI Linux build
+    ## Marked as 'Pending' on Linux for now because the test requires root privilege but we cannot do it now in our Azure DevOps Linux build
```
Comment (adityapatwardhan) [inline test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: Is this true in Azure DevOps? Should we try enabling these two tests?

Comment (RDIL) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: We can try

Comment (RDIL) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: I’ve temporarily made it run on linux so we can check.

**Comment by RDIL** [comment]
**Update: it works normally in Azure so we can delete this comment entirely!**

**Comment by RDIL** [comment]
@adityapatwardhan @iSazonov please re-review.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by iSazonov** [comment]
@RDIL Please look CI-Linux errors.

**Comment by RDIL** [comment]
I guess the comment is still right

**Comment by iSazonov** [comment]
Yes.

**Comment by adityapatwardhan** [comment]
@RDIL Can you add `RequireSudoOnUnix` tag to both describes?

**Comment by RDIL** [comment]
@adityapatwardhan I believe I did it, not sure.  
(I'm new to PowerShell syntax)

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 [hunk 2]
```diff
@@ -158,8 +158,8 @@ Describe "PowerShellGet - Module tests (Admin)" -tags @('Feature', 'RequireAdmin
         Remove-InstalledModules
     }
 
-    ## Marked as 'Pending' on Linux for now because the test requires root privilege but we cannot do it now in our Travis CI Linux build
-    It "Should install a module correctly to the required location with AllUsers scope" -Pending:$IsLinux {
+    ## Marked as 'Pending' on Linux for now because the test requires root privilege but we cannot do it now in our Azure DevOps Linux build
+    It "Should install a module correctly to the required location with AllUsers scope" -Tags @('RequireSudoOnUnix') {
```
Comment (iSazonov) [inline test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: Tags is for Describe block.

Comment (RDIL) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: How would I add tags here?

Comment (iSazonov) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: Please remove the tag from "It".
Line 148 should be:
```
-Tags @('Feature','RequireAdminOnWindows', 'RequireSudoOnUnix'
```

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 [hunk 3]
```diff
@@ -158,7 +158,7 @@ Describe "PowerShellGet - Module tests (Admin)" -tags @('Feature', 'RequireAdmin
         Remove-InstalledModules
     }
 
-    ## Marked as 'Pending' on Linux for now because the test requires root privilege but we cannot do it now in our Travis CI Linux build
+    ## Marked as 'Pending' on Linux for now because the test requires root privilege but we cannot do it now in our Azure DevOps Linux build
     It "Should install a module correctly to the required location with AllUsers scope" -Pending:$IsLinux {
```
Comment (iSazonov) [inline test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: Plan was to remove the pending here and below? Yes?

Comment (RDIL) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: Yes however it needs root.  Aditya said to add tags but I'm not sure how right here so I just reverted it for now.

Comment (RDIL) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: Fixed!

File: test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1 [hunk 4]
```diff
@@ -25,11 +25,14 @@ if(IsInbox)
 {
     $script:ProgramFilesPSPath = Microsoft.PowerShell.Management\Join-Path -Path $env:ProgramFiles -ChildPath "WindowsPowerShell"
 }
-elseif(IsCoreCLR){
-    if(IsWindows) {
+elseif(IsCoreCLR)
+{
```
Comment (iSazonov) [inline test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: Don't use C# style for PowerShell scripts.
Please revert the brace style changes. Below too.

Comment (RDIL) [reply test/powershell/Modules/PowerShellGet/PowerShellGet.Tests.ps1]: Fixed

**Comment by RDIL** [comment]
Everything is now fixed, ready for review/merge.

**Review by iSazonov — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
Restarted CI for Linux

**Comment by RDIL** [comment]
Thanks

**Comment by adityapatwardhan** [comment]
@RDIL can you update the PR description and context please?

**Comment by RDIL** [comment]
Sure

**Comment by RDIL** [comment]
Done

**Comment by adityapatwardhan** [comment]
@RDIL Thank you for your contribution

**Pull request merged**